scripts/schemaGenerator.py

```diff
@@ -64,47 +79,60 @@ def get_options_list_for_enum(field):
 
 def generalize_regex_pattern(field):
     """Generates the pattern for schema based on the type of field.
-    - For type : singleSelect and dynamicSelectForm, the pattern is generated by iterating over options.
-    - For other types,
-        - If the field contains regex, then regex is the pattern; it gets prefixed with a default prefix if regex does not have it.
-        - Else, the default prefix is appended with ^(.{0,100}).
-
+    
     Args:
         field (object): Individual field in ui-config, includes properties such as label, type, name, regex etc.
 
     Returns:
         string: generated pattern for the field.
     """
-    defaultSubPattern = "(^\\{\\{.*\\|\\|(.*)\\}\\}$)"
-    defaultEnvPattern = "(^env[.].+)"
-    pattern = ""
-    if "regex" in field:
-        pattern = field["regex"]
-        if defaultSubPattern not in pattern and (
-            ("value" not in field or field["value"] != "purpose")
-            and ("configKey" not in field or field["configKey"] != "purpose")
-        ):
-            pattern = "|".join([defaultSubPattern, pattern])
-        if defaultEnvPattern not in pattern and (
-            ("value" not in field or field["value"] != "purpose")
-            and ("configKey" not in field or field["configKey"] != "purpose")
-        ):
-            indexToPlace = pattern.find(defaultSubPattern) + len(defaultSubPattern)
-            pattern = (
-                pattern[:indexToPlace]
-                + "|"
-                + defaultEnvPattern
-                + pattern[indexToPlace:]
-            )
-    # TODO: we should not use a case here for the individual properties. Just pass the desired pattern as regex property
-    #  in ketch purpose fields and delete next case
-    elif ("value" in field and field["value"] == "purpose") or (
+    def extract_generic_fallback(regex):
+        # Split on |, return the part that matches ^(.{N,M})$ for any N, M
+        parts = [p.strip() for p in regex.split('|')]
+        for part in parts:
+            if re.match(GENERIC_FALLBACK_PATTERN, part):
+                return part
+        # fallback to default if not found
+        return DEFAULT_GENERIC_FALLBACK
+
+    if "regex" in field and field["regex"] and field["regex"].strip():
+        existing_pattern = field["regex"].strip()
+        generic_fallback = extract_generic_fallback(existing_pattern)
+        # Check if the pattern already contains redundant parts
+        # If it has both dynamic config/env patterns AND a generic fallback, it's redundant
+        has_dynamic_config = has_explicit_dynamic_config_pattern(existing_pattern)
```

Comment:
why do we need all the handling of dynamic config & env patterns still here ?
validating dynamic config/env patterns has been moved to config-be - so can db-config & schema.json focus only on business use-case based regex now ?

Reply:
but if remove from here UI won't allow them to be submitted, sometimes we use exact patterns like [0-9]+ then dynamic patterns will be blocked from UI so we need to add them explictly.